### PR TITLE
Add WinUI support

### PR DESCRIPTION
**Pull request by jeromelaban**
This PR supersedes https://github.com/mono/SkiaSharp/pull/1960.

**Description of Change**

Added views for Uno.WinUI https://github.com/mono/SkiaSharp/issues/1787

**Bugs Fixed**

**API Changes**

None

**Behavioral Changes**

None

**PR Checklist**

- [ ] Has tests (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: source/SkiaSharp.Views/SkiaSharp.Views.Shared/GlesInterop/Gles.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#if !__WATCHOS__ && !__WASM__ && !UNO_REFERENCE_API && !__MACCATALYST__
+﻿#if !__WATCHOS__ && !__WASM__ && (!UNO_REFERENCE_API || (NET6_0_OR_GREATER && (__IOS__ || __MACOS__)))
```
Comment (mattleibow) [inline source/SkiaSharp.Views/SkiaSharp.Views.Shared/GlesInterop/Gles.cs:1]: not: is this change required since it makes it harder to read.

File: samples/Basic/Uno/global.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "sdk": {
+    "version": "6.0.102"
```
Comment (mattleibow) [inline samples/Basic/Uno/global.json:3]: Is this still needed?

**Review by mattleibow — COMMENTED** [review]
Looking really good.

Just need the nuget stuff:
 - .nusepc
 - version in VERSIONS.txt
 - entry in TRACKED_NUGETS in build.cake

File: source/SkiaSharp.Views/SkiaSharp.Views.Shared/GlesInterop/Gles.cs (hunk 1 above)
Comment (jeromelaban) [reply source/SkiaSharp.Views/SkiaSharp.Views.Shared/GlesInterop/Gles.cs:1]: The `UNO_REFERENCE_API` is incorrectly defined for net6 in uno, and I don't want to force an update in SkiaSharp for this. I'll add a note to the related issue.

File: nuget/SkiaSharp.Views.Uno.WinUI.nuspec [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    <tags>ui uno xamarin graphics ios android catalyst windows winui macos cross-platform skiasharp</tags>
+
+    <!-- legal -->
+    <licenseUrl>https://go.microsoft.com/fwlink/?linkid=868514</licenseUrl>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+
+    <dependencies>
```
Comment (mattleibow) [inline nuget/SkiaSharp.Views.Uno.WinUI.nuspec:29]: What should happen if I install this package into a WinUI app? Should we add a net6.0-windows dependency here as well that does nothing or depends on the SkiaSharp.Views.WinUI? It may also need a `_._` file in the windows libs so that nothing gets installed.

File: samples/Basic/Uno/global.json (hunk 2 above)
Comment (jeromelaban) [reply samples/Basic/Uno/global.json:3]: It is not, let's remove it.

File: nuget/SkiaSharp.Views.Uno.WinUI.nuspec (hunk 3 above)
Comment (jeromelaban) [reply nuget/SkiaSharp.Views.Uno.WinUI.nuspec:29]: Indeed, it's useful because of the `netstandard2.0` target, I'll add it back similar to the way it is in the SkiaSharp.Views.Uno.

**Review by mattleibow — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]
Been a long one and it is a big one. But, it is a green one!

**Comment by mattleibow** [comment]
Merged in 7b3e8f3375d8114c473f87803627ff6a17519130

**Comment by mattleibow** [comment]
Had to do a local merge and not get the nice purple because I don't have push permissions. But, the credits are in the commit and this PR lives on!

Thanks so much for the work on this, really glad to get even more SkiaSharp view support!

**Comment by TopperDEL** [comment]
When will that Nuget be available? Is this bound to a release of something else?

**Comment by mattleibow** [comment]
It should be coming out soon. I am just trying to get CI to build it as we type. Something just gave up and dies so now it runs out of disk space on CI... So it says. Hopefully this weekend or next week.

**Comment by TopperDEL** [comment]
That's nice! Thanks for the update! :)

**Comment by TopperDEL** [comment]
Any news on this one @mattleibow?

**Comment by TopperDEL** [comment]
The package is there! Thank you! :)

**Comment by mattleibow** [comment]
Yeah, sorry about not updating this issue. I pushed the nuget and then started on some international flights.

**Comment by TopperDEL** [comment]
All fine! I just polled nuget.org until I found the package and have been the first one to download it. 🤣
Thanks for your work and your update on this!
